### PR TITLE
Fixed #34701 -- Added support for NULLS [NOT] DISTINCT on PostgreSQL 15+.

### PR DESCRIPTION
**Pull request by charettes**

### CODE REVIEW

File: django/db/backends/base/schema.py [hunk 1]
```diff
@@ -1675,6 +1675,37 @@ def _deferrable_constraint_sql(self, deferrable):
         if deferrable == Deferrable.IMMEDIATE:
             return " DEFERRABLE INITIALLY IMMEDIATE"
 
+    def _unique_index_nulls_distinct_sql(self, nulls_distinct):
+        if nulls_distinct is False:
+            return " NULLS NOT DISTINCT"
+        elif nulls_distinct is True:
+            return " NULLS DISTINCT"
+        return ""
```
Comment (shangxiao) [inline django/db/backends/base/schema.py:1683]: This is just a preference thing but normally I'd put the positive case first then negative to aide readability. Is there a Django preference?

Some people also like to put the default in an else to align the returns.

eg:

```
if someting is true:
    ...
elif something is false:
    ...
else:
    ...
```

**Review by shangxiao — APPROVED** [review]
If only adding `ROLLUP` support to group by was this easy 🎉

File: django/db/backends/base/schema.py (hunk 1 above)
Comment (ngnpope) [reply django/db/backends/base/schema.py:1683]: I'd prefer `elif` → `if` on the second line 😁

**Review by felixxm — COMMENTED** [review]
@charettes Thanks :+1: Looks fantastic :ok_hand: 

I pushed small edits, e.g. added `nulls_distinct` to the `UniqueConstraint.__eq__()` with tests.

**Pull request merged**